pkg/backoff/exponential.go

```diff
@@ -38,22 +45,51 @@ type ExponentialBackOff struct {
 }
 
 // NewExponentialBackOff creates an istio wrapped ExponentialBackOff.
+// By default, it never stops.
 func NewExponentialBackOff(initFuncs ...func(off *ExponentialBackOff)) BackOff {
-	b := &ExponentialBackOff{}
+	b := ExponentialBackOff{}
 	b.ExponentialBackOff = backoff.NewExponentialBackOff()
+	b.ExponentialBackOff.MaxElapsedTime = 0
 	for _, fn := range initFuncs {
-		fn(b)
+		fn(&b)
 	}
 	b.Reset()
 	return b
 }
 
-const MaxDuration = 1<<63 - 1
-
-func (b *ExponentialBackOff) NextBackOff() time.Duration {
+func (b ExponentialBackOff) NextBackOff() time.Duration {
 	duration := b.ExponentialBackOff.NextBackOff()
 	if duration == b.Stop {
-		return MaxDuration
+		return b.ExponentialBackOff.MaxInterval
 	}
 	return duration
 }
+
+func (b ExponentialBackOff) Reset() {
+	b.ExponentialBackOff.Reset()
+}
+
+// RetryWithContext tries the operation until it does not return error, BackOff stops,
```

Comment:
```suggestion
// RetryWithContext tries the operation until it does not return error, BackOff hits MaxElapsedTime,
```

To meet current implementation.

however, IMO we should not expose MaxElapsedTime at all to users, and instead have them use only the context. 2 benefits:
* only 1 way to cancel the retry instead of 2
* only 1 way to set max internal instead of 2

Reply:
This is because we are using backoff differently, NextBackOff in some places, retry in some other places